server.go

```diff
@@ -429,7 +431,9 @@ type Server struct {
 	mu   sync.Mutex
 	open int32
 	stop int32
-	done chan struct{}
+
+	done      chan struct{}
+	closeDone func()
```

Comment:
To be honest, `fasthttputil.OnceFunc` is a bit of overkill. I would just add a `sync.Once` here instead of `closeDone` and instead of calling `s.closeDone()` you do `s.closeDone.Once(func() { close(s.done) })`.

Reply:
ok